<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MIDIQtClass</class>
 <widget class="QMainWindow" name="MIDIQtClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>601</width>
    <height>338</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GCVillager的MIDI电子琴</string>
  </property>
  <property name="windowIcon">
   <iconset resource="MIDIQt.qrc">
    <normaloff>:/MIDIQt/icon.ico</normaloff>:/MIDIQt/icon.ico</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">#MIDIQtClass{
border-image:url(background.jpg);
}
QComboBox{
background-color:rgba(255,255,255,0.4);
}
QPushButton{
background-color:rgba(225,225,225,0.45);
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string notr="true">#centralWidget{background-color:transparent;}</string>
   </property>
   <widget class="QComboBox" name="comboMajorMinor">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>61</width>
      <height>22</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>大小调选择</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>30</y>
      <width>54</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>调式选择</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboScale">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>50</y>
      <width>61</width>
      <height>22</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="maxVisibleItems">
     <number>12</number>
    </property>
   </widget>
   <widget class="QComboBox" name="comboIns">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>110</y>
      <width>451</width>
      <height>22</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>90</y>
      <width>54</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>乐器选择</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinDelay">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>50</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::WheelFocus</enum>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:rgb(255,252,243);</string>
    </property>
    <property name="maximum">
     <number>5000</number>
    </property>
    <property name="singleStep">
     <number>100</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>30</y>
      <width>181</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>延时停止时长(滚动滚轮改变)</string>
    </property>
   </widget>
   <widget class="QSlider" name="sliderVolume">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>50</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>54</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>乐器种类</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboInsType">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>111</width>
      <height>22</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="labelCurVolume">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>30</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>当前音量：100</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>30</y>
      <width>54</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>音量调节</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>50</y>
      <width>75</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string>保存配置</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonStartRecord">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>101</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>开始记录</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonStopSave">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>160</y>
      <width>91</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>停止记录并保存</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonStopGiveUp">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>160</y>
      <width>91</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>停止记录并放弃</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonOpenFile">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>101</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>打开记录文件</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonStartReplay">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>220</y>
      <width>91</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>(重新)开始播放</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonStopReplay">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>220</y>
      <width>91</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>停止播放</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelNowPlaying">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>200</y>
      <width>581</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>当前打开文件：无</string>
    </property>
   </widget>
   <widget class="QSlider" name="sliderReplayVolume">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>250</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>250</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>回放音量调节</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelReplayVolume">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>250</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>当前音量：100</string>
    </property>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="MIDIQt.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>comboMajorMinor</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>changeScale()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>37</x>
     <y>80</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboScale</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>changeScale()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>118</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>118</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboInsType</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>changeInsType()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>148</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboIns</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>changeIns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>562</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>547</x>
     <y>80</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderVolume</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>changeVolume()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>293</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>293</x>
     <y>74</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinDelay</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>changeDelay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>445</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>saveCfg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>571</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>32</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonStartRecord</sender>
   <signal>clicked()</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>startRecord()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>78</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonStopSave</sender>
   <signal>clicked()</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>stopRecordSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>191</x>
     <y>197</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonStopGiveUp</sender>
   <signal>clicked()</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>stopRecordGiveUp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonOpenFile</sender>
   <signal>clicked()</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>openRecord()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>248</y>
    </hint>
    <hint type="destinationlabel">
     <x>67</x>
     <y>267</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonStartReplay</sender>
   <signal>clicked()</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>startReplay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>169</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonStopReplay</sender>
   <signal>clicked()</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>stopReplay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>286</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderReplayVolume</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MIDIQtClass</receiver>
   <slot>changeVolume()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>271</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>286</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changeScale()</slot>
  <slot>changeInsType()</slot>
  <slot>changeIns()</slot>
  <slot>changeVolume()</slot>
  <slot>changeDelay()</slot>
  <slot>saveCfg()</slot>
  <slot>startRecord()</slot>
  <slot>stopRecordSave()</slot>
  <slot>stopRecordGiveUp()</slot>
  <slot>openRecord()</slot>
  <slot>startReplay()</slot>
  <slot>stopReplay()</slot>
 </slots>
</ui>
